### PR TITLE
Add subclasses to autonomous OpMode

### PR DESCRIPTION
**Pull request by avanibansal**
Add subclasses (depending on team color and side) to autonomous OpMode

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java [hunk 1]
```diff
@@ -29,15 +29,29 @@ public enum From {
         START, CONE_STACK
     }
 
+    public enum TeamState {
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java:32]: Let's break this into two `enums` -- `TeamColor` and `TeamSide`

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java [hunk 2]
```diff
@@ -29,15 +29,29 @@ public enum From {
...
     private State mCurrentState;                         // Current State Machine State.
-
+    protected TeamState teamState;
+    private int sign;
 
     /**
      * Initializes State Machine
      */
-    public void init() {
+    public void init(TeamState teamState) {
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java:47]: Change params to `color` and `side` and (create and) set two fields to keep track of them

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java [hunk 3]
```diff
@@ -87,8 +101,15 @@ public void loop() {
                 };
                 break;
             case ALIGN_WITH_CONE:
-                if (align(PixyCam.BLUE, CONE_WIDTH)) {
-                    newState(State.INTAKE_CONE);
+                // if blue team, intake blue cone
+                if (teamState == TeamState.BLUE_LEFT || teamState == TeamState.BLUE_RIGHT) {
```
Comment (zerbash) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java:105]: Both the `if` and the `else` are almost the same statement. Consider something like this:

```
int color = TeamColor.BLUE ? PixyCam.BLUE : PixyCam.RED;
if (align(color, CONE_WIDTH)) {
    newState(State.INTAKE_CONE);
}
```

**Review by zerbash — CHANGES_REQUESTED** [review]
@ExternalYelu @C0de555

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java (hunk 1 above)
Comment (ExternalYelu) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java:32]: got it, ill do it during break

**Pull request closed without merging**